### PR TITLE
Test multiple url patterns and skip pattern all defined in file

### PR DESCRIPTION
**Pull request by pavolloffay**

### CODE REVIEW

File: opentracing-spring-web-autoconfigure/src/test/java/io/opentracing/contrib/spring/web/autoconfig/ServerTracingAutoConfigurationTest.java [hunk 1]
```diff
@@ -62,26 +82,35 @@ public MockTracer tracer() {
     @Test
     public void testRequestIsTraced() {
         testRestTemplate.getForEntity("/hello", String.class);
-        Awaitility.await().until(reportedSpansSize(), IsEqual.equalTo(2));
-
-        assertThat(mockTracer.finishedSpans()).hasSize(2);
+        Awaitility.await().until(reportedSpansSize(), IsEqual.equalTo(1));
+        assertThat(mockTracer.finishedSpans()).hasSize(1);
```
Comment (objectiser) [inline opentracing-spring-web-autoconfigure/src/test/java/io/opentracing/contrib/spring/web/autoconfig/ServerTracingAutoConfigurationTest.java]: What is the reason for the change in span/trace number from 2 to 1?

File: opentracing-spring-web-autoconfigure/src/test/java/io/opentracing/contrib/spring/web/autoconfig/ServerTracingAutoConfigurationTest.java [hunk 2]
```diff
@@ -62,26 +82,35 @@ public MockTracer tracer() {
...
         assertThat(mockTracer.finishedSpans()).hasSize(0);
         assertThat(Mockito.mockingDetails(mockDecorator1).getInvocations()).hasSize(0);
+        assertThat(Mockito.mockingDetails(mockDecorator2).getInvocations()).hasSize(0);
     }
 
     @Test
     public void testDecoratedProviderIsUsed() {
         testRestTemplate.getForEntity("/hello", String.class);
-        Awaitility.await().until(reportedSpansSize(), IsEqual.equalTo(2));
+        Awaitility.await().until(reportedSpansSize(), IsEqual.equalTo(1));
```
Comment (objectiser) [inline opentracing-spring-web-autoconfigure/src/test/java/io/opentracing/contrib/spring/web/autoconfig/ServerTracingAutoConfigurationTest.java]: Same as above - just wondering why 2 down to 1?

File: opentracing-spring-web-autoconfigure/src/test/java/io/opentracing/contrib/spring/web/autoconfig/ServerTracingAutoConfigurationTest.java [hunk 3]
```diff
@@ -62,26 +82,35 @@ public MockTracer tracer() {
...
 
-        assertThat(mockTracer.finishedSpans()).hasSize(2);
+        assertThat(mockTracer.finishedSpans()).hasSize(1);
         assertThat(Mockito.mockingDetails(mockDecorator1).getInvocations()).hasSize(2);
```
Comment (objectiser) [inline opentracing-spring-web-autoconfigure/src/test/java/io/opentracing/contrib/spring/web/autoconfig/ServerTracingAutoConfigurationTest.java]: Is this dependent on order tests invoked? Should the mocks be reset just in case.

File: opentracing-spring-web-autoconfigure/src/test/java/io/opentracing/contrib/spring/web/autoconfig/ServerTracingAutoConfigurationTest.java (hunk 1 above)
Comment (pavolloffay) [reply opentracing-spring-web-autoconfigure/src/test/java/io/opentracing/contrib/spring/web/autoconfig/ServerTracingAutoConfigurationTest.java]: Previous there were 2 spans. There wasn't a controller so the first span is for missing controller and second span was for error controller.

File: opentracing-spring-web-autoconfigure/src/test/java/io/opentracing/contrib/spring/web/autoconfig/ServerTracingAutoConfigurationTest.java (hunk 3 above)
Comment (pavolloffay) [reply opentracing-spring-web-autoconfigure/src/test/java/io/opentracing/contrib/spring/web/autoconfig/ServerTracingAutoConfigurationTest.java]: it's probably not because it works.

Comment (objectiser) [reply opentracing-spring-web-autoconfigure/src/test/java/io/opentracing/contrib/spring/web/autoconfig/ServerTracingAutoConfigurationTest.java]: might still be best to reset them, to avoid any unexpected issues later, but I'll leave that up to you.

Comment (pavolloffay) [reply opentracing-spring-web-autoconfigure/src/test/java/io/opentracing/contrib/spring/web/autoconfig/ServerTracingAutoConfigurationTest.java]: Docs says that reset is applied on `MockBeans` automatically after the test.
